src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,543 @@
+
+    if (!declaredParameter) {
+      throw new Error(`Empty parameter slot: ${textRaw}`);
+    }
+
+    // We need to find out if this parameter is optional or not. We can tell this
+    //  if we're wrapped in brackets, so let's look for them.
+
+    let pos;
+    for (pos = 0; pos < declaredParameter.length; pos++) {
+      const levelChange = optionalCharDict[declaredParameter[pos]];
+
+      if (levelChange === undefined) {
+        break;
+      }
+
+      optionalDepth += levelChange;
+    }
+
+    // Cut off any trailing brackets
+    declaredParameter = declaredParameter.substring(pos);
+
+    const isParameterOptional = optionalDepth > 0;
+
+    for (pos = declaredParameter.length - 1; pos >= 0; pos--) {
+      const levelChange = optionalCharDict[declaredParameter[pos]];
+
+      if (levelChange === undefined) {
+        break;
+      }
+
+      optionalDepth += levelChange;
+    }
+
+    // Cut off any leading brackets
+    declaredParameter = declaredParameter.substring(0, pos + 1);
+
+    // Default value of this parameter in the method's declaration
+    let defaultValue;
+
+    const equalSignPos = declaredParameter.indexOf('=');
+    if (equalSignPos !== -1) {
+      // We have a default value, save it and then cut it off of the signature
+      defaultValue = declaredParameter.substring(equalSignPos, 1).trim();
+      declaredParameter = declaredParameter.substring(0, equalSignPos);
+    }
+
+    let parameter = rawParameters[i];
+    if (!parameter || declaredParameter !== parameter.name) {
+      // If we're here then the method likely has shared signatures
+      //  Something like, `new Console(stdout[, stderr][, ignoreErrors])` and
+      //  `new Console(options)`
+      parameter = undefined;
+
+      // Try finding a parameter this is being shared with
+      for (const otherParam of rawParameters) {
+        if (declaredParameter === otherParam.name) {
+          // Found a matching one
+          parameter = otherParam;
+          break;
+        } else if (otherParam.options) {
+          // Found a matching one in the parameter's options
+          for (const option of otherParam.options) {
+            if (declaredParameter === option.name) {
+              parameter = Object.assign({}, option);
+              break;
+            }
+          }
+        }
+      }
+
+      if (!parameter) {
+        // Couldn't find the shared one
+        if (declaredParameter.startsWith('...')) {
+          parameter = { name: declaredParameter };
+        } else {
+          throw new Error(
+            `Invalid param "${declaredParameter}"\n` + ` > ${textRaw}`
+          );
+        }
+      }
+    }
+
+    if (isParameterOptional) {
+      parameter.optional = true;
+    }
+
+    if (defaultValue) {
+      parameter.default = defaultValue;
+    }
+
+    signature.params.push(parameter);
+  });
+
+  return signature;
+}
+
+/**
+ * @param {Array<import('mdast').PhrasingContent>} nodes
+ */
+function textJoin(nodes) {
```

Comment:
Are you just making this Markdown-ish? We already have this https://github.com/nodejs/api-docs-tooling/blob/main/src/utils/unist.mjs#L12